nixos/modules/services/web-apps/firefly-iii.nix

```diff
@@ -0,0 +1,438 @@
+{ pkgs, config, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.firefly-iii;
+  db = cfg.database;
+  mail = cfg.mail;
+
+  user = cfg.user;
+  group = cfg.group;
+
+  firefly-iii = pkgs.firefly-iii.override {
+    dataDir = cfg.dataDir;
+  };
+
+  defaultUser = "firefly-iii";
+  defaultGroup = defaultUser;
+
+  tlsEnabled = cfg.nginx.addSSL || cfg.nginx.forceSSL || cfg.nginx.onlySSL || cfg.nginx.enableACME;
+
+  # shell script for local administration
+  artisan = pkgs.writeScriptBin "firefly-iii" ''
+    #! ${pkgs.runtimeShell}
+    cd ${firefly-iii}
+    sudo=exec
+    if [[ "$USER" != ${user} ]]; then
+      sudo='exec /run/wrappers/bin/sudo -u ${user}'
+    fi
+    $sudo ${pkgs.php}/bin/php artisan $*
+  '';
+in {
+
+  options.services.firefly-iii = {
+
+    enable = mkEnableOption "Firefly III";
+
+    dataDir = mkOption {
+      description = "Firefly III data directory";
+      default = "/var/lib/firefly-iii";
+      type = types.path;
+    };
+
+    # App configuration
+    appURL = mkOption {
+      description = ''
+        The root URL that you want to host Firefly III on. All URLs in Firefly III will be generated using this value.
+      '';
+      default = "http${optionalString tlsEnabled "s"}://${cfg.hostname}";
+      defaultText = ''http''${optionalString tlsEnabled "s"}://''${cfg.hostname}'';
+      example = "https://example.com";
+      type = types.str;
+    };
+
+    appKeyFile = mkOption {
+      description = ''
+        A file containing the Laravel APP_KEY - a 32 character long,
+        base64 encoded key used for encryption where needed. Can be
+        generated with <code>head -c 32 /dev/urandom | base64</code>.
+      '';
+      example = "/run/keys/firefly-iii-appkey";
+      type = types.path;
+    };
+
+    # PHP
+    poolConfig = mkOption {
+      type = with types; attrsOf (oneOf [ str int bool ]);
+      default = {
```

Comment:
It might be a good idea to add the following lines to the default `poolConfig`:

```
"listen.owner" = "nginx";
"listen.group" = "nginx";
"listen.mode" = "0600";
```

Reply:
This should ideally clear up the issue @timhae has pointed out, I think.

Reply:
@savyajha , done. Thanks!